src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj.hidden === false) {
+      obj.health -= 50;
+    }
+
+    const filteredAnimal = Animal.alive.filter(({ health }) => health > 0);
+
+    Animal.alive = filteredAnimal;
```

Comment:
```suggestion
    Animal.alive = Animal.alive.filter(({ health }) => health > 0);
```
Also, it would be great to add a check if target's health is equal to zero to avoid extra method calls